src/graph_input.cpp

```diff
@@ -26,6 +26,20 @@ unsigned long read_ulong(char const* source, char** end)
     return value;
 }
 
+float read_float(char const* source, char** end)
+{
+    float const value = std::strtof(source, end);
+
+#if !defined(__clang__)
+    if (errno == ERANGE)
```

Comment:
should errno be set to 0 first, to make sure that we do not check for an old error?

Reply:
I've never seen anyone setting `errno` a-priori :thinking: The [cppreference](https://en.cppreference.com/w/cpp/error/errno) entry does not explain if it is good practice to set it to `0` after (not before) the comparison with `ERANGE` has succeeded.

Reply:
`errno` is set to `0` now after it was set to `ERANGE`.